Two joint driver

TheInfamousWayne opened the pull request:
added driver, demo and python binding files to run two joints using 1 motor-board

TheInfamousWayne commented:
> This PR seems to be incomplete, the new files are missing (e.g. `srcpy/py_two_joint.cpp` which is mentioned in the new CMakeLists.txt entry).

I'll add them and make a new PR. Unsure why it didn't upload. Also, will make the changes to CMakeLists.txt

luator commented:
No need to make a new PR, you can simply add it here.

luator commented on demos/demo_twojoint_print_position.py:
```diff
@@ -0,0 +1,27 @@
+home_offset = np.array([2.256, 2.2209])
+robot_data = two_joint.Data()
+robot_backend = blmc_robots.create_two_joint_backend("can6",
+						     home_offset,
```
Please fix indentation so all arguments are properly aligned.

luator commented on demos/demo_twojoint_print_position.py:
```diff
@@ -0,0 +1,27 @@
+
+
+robot_backend.initialize()
+print("4")
```
Looks like a debug print, if not needed anymore, please remove it.

luator commented on demos/demo_twojoint_print_position.py:
```diff
@@ -0,0 +1,27 @@
+print("4")
+# TODO this makes the application hang
+#t = finger.get_current_time_index()
+#print("t = %d" % t)
```
Please also remove the commented lines (also below in the loop)

luator commented on demos/demo_twojoint_robot.py:
```diff
@@ -0,0 +1,107 @@
+
+    robot_data = two_joint.Data()
+    robot_backend = blmc_robots.create_two_joint_backend("can6",
+							 home_offset,
```
indentation

luator commented on demos/demo_twojoint_robot.py:
```diff
@@ -0,0 +1,107 @@
+    go_to_zero(1000, 2000)
+
+    action = two_joint.Action(torque=np.ones(N_JOINTS) * 0.15)
+    robot.append_desired_action(action)
```
Why applying a non-zero torque for just one time step in between the position controllers? It seems like those two lines could be removed.

luator commented on include/blmc_robots/two_joint_driver.hpp:
```diff
@@ -0,0 +1,124 @@
+namespace blmc_robots
+{
+/**
+ * @brief Driver for a single joint.
```
It is still saying "single joint" here.

luator commented on include/blmc_robots/two_joint_driver.hpp:
```diff
@@ -0,0 +1,124 @@
+                                      true)
+    {
+        home_offset_rad_ << home_offset_rad;
+        // initial_position_rad_ << -2.256, -2.2209; // defaults to 0
```
Please remove this line completely if not needed.

luator commented on include/blmc_robots/two_joint_driver.hpp:
```diff
@@ -0,0 +1,124 @@
+        Vector vec;
+        vec << value, value;
+        return vec;
+    }
```
If I did not miss something, this method is not used here, so it can be removed.

luator commented on demos/demo_onejoint_print_position.py:
```diff
@@ -0,0 +1,26 @@
+
+# TODO this makes the application hang
+#t = finger.get_current_time_index()
+#print("t = %d" % t)
```
Sorry, I missed this the first time, the commented stuff here could also be removed

TheInfamousWayne commented:
**** FINAL CHANGES ****
Added files: demo_twojoint_robot.py (for left2right movement); demo_twojoint_print_position.py (for printing position of the two joints)
Added driver files and python bindings for 2 joint interface
Modified CMakeLists.txt for related changes
